airflow/providers/google/cloud/hooks/datafusion.py

```diff
@@ -67,6 +85,49 @@ def wait_for_operation(self, operation: Dict[str, Any]) -> Dict[str, Any]:
             raise AirflowException(operation["error"])
         return operation["response"]
 
+    def wait_for_pipeline_state(
+        self,
+        success_states: List[str],
+        pipeline_name: str,
+        faux_pipeline_id: str,
+        instance_url: str,
+        namespace: str = "default",
+        failure_states: List[str] = None,
+        timeout: int = 5 * 60,
+    ):
+        """
+        Pools pipeline state and raises an exception if the state is one of
+        `failure_states` or the operation timeouted.
+        """
+        failure_states = failure_states or FAILURE_STATES
+        start_time = monotonic()
+        current_state = None
+        while True:
```

Comment:
While true seems infinite loop prone. Any reason to not merge this with the below if?
```python3
    while monotonic - start_time > timepout:
        ...
    raise AirflowException(....)
```